eZ/Publish/Core/REST/Server/Input/Parser/Criterion/UserMetadata.php

```diff
@@ -30,6 +32,28 @@ class UserMetadata extends BaseParser
      */
     public function parse(array $data, ParsingDispatcher $parsingDispatcher)
     {
-        throw new \Exception('@todo implement');
+        if (!isset($data['UserMetadataCriterion'])) {
+            throw new Exceptions\Parser('Invalid <UserMetadataCriterion> format');
+        }
+
+        if (!isset($data['UserMetadataCriterion']['Target'])) {
+            throw new Exceptions\Parser('Invalid <Target> format');
+        }
+
+        $target = $data['UserMetadataCriterion']['Target'];
+
+        if (!isset($data['UserMetadataCriterion']['Value'])) {
+            throw new Exceptions\Parser('Invalid <Value> format');
+        }
+
+        if (!in_array(gettype($data['UserMetadataCriterion']['Value']), ['integer', 'string', 'array'])) {
+            throw new Exceptions\Parser('Invalid <Value> format');
+        }
+
+        $value = is_array($data['UserMetadataCriterion']['Value'])
```

Comment:
@pspanja Now I don't see how it can happen, either. @Nattfarinn, can you shed light on it please?